var/spack/repos/builtin/packages/llvm/llvm6.0.0+py3.7.patch

```diff
@@ -0,0 +1,37 @@
+From ecdefed7f6ba11421fe1ecc6c13a135ab7bcda73 Mon Sep 17 00:00:00 2001
+From: Pavel Labath <labath@google.com>
+Date: Mon, 23 Jul 2018 11:37:36 +0100
+Subject: [PATCH] Fix PythonString::GetString for >=python-3.7
+
+
+The return value of PyUnicode_AsUTF8AndSize is now "const char *".
+---
+ .../Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp  | 6 ++++--
+ 1 file changed, 4 insertions(+), 2 deletions(-)
+
+diff --git a/tools/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp b/tools/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
+index 6a9d57d5a..94f16b2c7 100644
+--- a/tools/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
++++ b/tools/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
+@@ -399,14 +399,16 @@ llvm::StringRef PythonString::GetString() const {
+     return llvm::StringRef();
+ 
+   Py_ssize_t size;
+-  char *c;
++  const char *data;
+ 
+ #if PY_MAJOR_VERSION >= 3
+   data = PyUnicode_AsUTF8AndSize(m_py_obj, &size);
```

Comment:
@flxmr 

This appears broken and the linked thread https://bugs.llvm.org/show_bug.cgi?id=38233 mentions that the original patch missed this so:

* You may be able to use 1 patch for both versions
* The bug report mentions a commit. The commit refers to the SVN repo but the github repo has `https://github.com/llvm-mirror/lldb/commit/b032b915ad7eea93e06d571f1a3ca712cd203b9e`, if you add `.patch` onto that, you can refer to it as a Spack URL patch

Reply:
Actually reading this again, I may be wrong that one patch can be used for both versions, in particular if `data = PyUnicode_AsUTF8AndSize(m_py_obj, &size);` is in `6.0.0` (that would be strange though, especially without the patch to create the instantiation of `const char *data`).

However, regarding the second point, the other patch `var/spack/repos/builtin/packages/llvm/llvm6.0.1+py3.7.patch` looks like `https://github.com/llvm-mirror/lldb/commit/b032b915ad7eea93e06d571f1a3ca712cd203b9e` and it would be preferred to use the URL patch.